lib/vector-core/src/transform/config.rs

```diff
@@ -15,10 +16,35 @@ pub enum ExpandType {
     Serial,
 }
 
+#[derive(Debug)]
+pub struct TransformContext {
+    pub globals: GlobalOptions,
+    pub enrichment_tables: EnrichmentTables,
+}
```

Comment:
Does it make sense to limit enrichment tables to transforms at this stage? Or, if we do, would it be relatively straightforward to expose them to sources/sinks in the future?

I'm thinking of some "transform" level stuff we've moved to sources, e.g. some of the codec stuff, that might at some point benefit from accessing enrichment tables.

Reply:
It should be easy to expose the sources/sinks in future. `EnrichmentTables` should be cheaply cloned and still point to the same underlying data structure. So when we need to pass to a source/sink, we can just clone and add to the appropriate `build` function.